### ISSUE TITLE
Bug or feature? Port sharing among processes using httplib

### ISSUE THREAD

**Issue by You-NeverKnow** [yhirose/cpp-httplib#250]
So when I run two programs compiled with `httplib.h` on the same port, my Ubuntu machine does NOT throw a "Port already in use" error. Instead, they dynamically share the port address.

I can simultaneously run binaries generated by both the files below:
```
// http-server1 
#include <iostream>
#include "httplib.h"

using namespace httplib;
int main (int argc, char *argv[]) {
    Server server;
    server.Get("/", [](const Request& req, Response& res) {
        res.set_content("Hello, Dimension C-137!", "text/plain");
    });

    std::cout << "Serving at localhost:3000/" << '\n';
    server.listen("localhost", 3000);
}
```

```
// http-server2
#include <iostream>
#include "httplib.h"

using namespace httplib;
int main (int argc, char *argv[]) {
    Server server;
    server.Get("/", [](const Request& req, Response& res) {
        res.set_content("Hello, World0!", "text/plain");
    });

    std::cout << "Serving at localhost:3000/" << '\n';
    server.listen("localhost", 3000);
}
```
And if I do get requests on them, the load seems to be evenly divided amongst them.
```
# Running from my Python shell with requests library
In [3]: for _ in range(15):
   ...:     r = requests.get("http://localhost:3000")
   ...:     print(r.content)
   ...:     
b'Hello, Dimension C-137!'
b'Hello, Dimension C-137!'
b'Hello, World0!'
b'Hello, Dimension C-137!'
b'Hello, Dimension C-137!'
b'Hello, World0!'
b'Hello, Dimension C-137!'
b'Hello, Dimension C-137!'
b'Hello, Dimension C-137!'
b'Hello, World0!'
b'Hello, Dimension C-137!'
b'Hello, Dimension C-137!'
b'Hello, World0!'
b'Hello, Dimension C-137!'
b'Hello, Dimension C-137!'
```

Is this intended?

**Comment by yhirose** [comment]
@You-NeverKnow, thank you for the report! It's actually an intentional behavior.

https://github.com/yhirose/cpp-httplib/blob/5f32c424c2d819a74bf00851b1fb5e1b28a88e57/httplib.h#L1253-L1256

`SO_REUSEPORT` causes this behavior. (This is available since Linux 3.9.)

For more information about this option, you can check the following web pages:
https://github.com/torvalds/linux/commit/c617f398edd4db2b8567a28e899a88f8f574798d
https://n0p.me/portsharding/
https://github.com/yhirose/cpp-httplib/issues/179

If you don't want this behavior in your situation, please remove or comment out the line.
Hope it helps.

**Comment by You-NeverKnow** [comment]
Wow. I wasn't aware of this. Thanks for the prompt response.
